packages/wdio-devtools-service/src/auditor.js

```diff
@@ -0,0 +1,131 @@
+import weightedMean from 'weighted-mean'
+
+import Diagnostics from 'lighthouse/lighthouse-core/audits/diagnostics'
+import MainThreadWorkBreakdown from 'lighthouse/lighthouse-core/audits/mainthread-work-breakdown'
+import Metrics from 'lighthouse/lighthouse-core/audits/metrics'
+import TTFBMetric from 'lighthouse/lighthouse-core/audits/time-to-first-byte'
+
+import { quantileAtValue } from './utils'
+import logger from '@wdio/logger'
+
+const log = logger('Auditor')
+
+/**
+ * metric scoring to calculate Lighthouse Performance Score
+ * contains: [media, falloff]
+ * see https://docs.google.com/spreadsheets/d/1_iDW0B870vZF6dAhf1Icdher0gX_KFxCd6Df0_fZ6do/edit#gid=283330180
+ */
+const METRIC_SCORING = {
+    FMP: [4000, 1600], // first meaningful paint
+    FI: [10000, 1700], // first interactive
+    CI: [10000, 1700], // consistently interactive
+    SI: [5500, 1250], // speed index
+    EIL: [100, 50] // estimated input latency
+}
+
+const WEIGHT_WITHIN_CATEGORY = {
+    FMP: 5,
+    FI: 5,
+    CI: 5,
+    SI: 1,
+    EIL: 1
+}
+
+const SHARED_AUDIT_CONTEXT = {
+    settings: { throttlingMethod: 'devtools' },
+    LighthouseRunWarnings: false,
+    computedCache: new Map()
+}
+
+export default class Auditor {
+    constructor (traceLogs, devtoolsLogs) {
+        this.devtoolsLogs = devtoolsLogs
+        this.traceLogs = traceLogs
+        this.url = traceLogs.pageUrl
+        this.loaderId = traceLogs.loaderId
+    }
+
+    _audit (AUDIT, params = {}) {
+        const auditContext = {
+            options: Object.assign({}, AUDIT.defaultOptions),
+            ...SHARED_AUDIT_CONTEXT
+        }
+
+        try {
+            return AUDIT.audit({
```

Comment:
styling question (not even suggestion): is it good to have multiple returns this way? Or it is better to have `result` variable, assign value to it and return in the very end?

Reply:
I prefer having the `result` variable

Reply:
as that variable wouldn't be used anyway, I prefer to return directly